lib/manageiq/api/common/application_controller_mixins/exception_handling.rb

```diff
@@ -0,0 +1,37 @@
+module ManageIQ
+  module API
+    module Common
+      module ApplicationControllerMixins
+        module ExceptionHandling
+          def self.included(other)
+            other.rescue_from(StandardError, RuntimeError) do |exception|
+              errors, status = collected_ipp_errors(exception)
+
+              render :json => errors, :status => status
+            end
+          end
+
+          def collected_ipp_errors(exception, default_code = 400)
+            errors = []
+            og_exception = exception
+
+            until exception.nil?
+              code = exception.respond_to?(:code) ? exception.code : exception.class.to_s
+              errors << {:status => code, :detail => exception.class.to_s + ": " + exception.message}
+              exception = exception.cause
+            end
+
+            # Overwrite the first error since that one most likely doen't have a code
+            errors.first[:status] = if ActionDispatch::ExceptionWrapper.rescue_responses.key?(og_exception.class.to_s)
```

Comment:
Is there value moving this all into a method ( `assign_default_code` ) I wonder if it could return:
```
 [{:errors => errors}, errors.first[:status]]
```